api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java

```diff
@@ -94,20 +94,22 @@ Schema createSchema(NameIdentifier ident, String comment, Map<String, String> pr
    * @param changes The metadata changes to apply.
    * @return The altered schema.
    * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws UnsupportedOperationException If the catalog does not support altering a schema.
    */
-  Schema alterSchema(NameIdentifier ident, SchemaChange... changes) throws NoSuchSchemaException;
+  Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException, UnsupportedOperationException;
 
   /**
    * Drop a schema from the catalog. If cascade option is true, recursively drop all objects within
    * the schema.
    *
-   * <p>If the catalog implementation does not support this operation, it may throw {@link
-   * UnsupportedOperationException}.
-   *
    * @param ident The name identifier of the schema.
    * @param cascade If true, recursively drop all objects within the schema.
    * @return True if the schema exists and is dropped successfully, false otherwise.
    * @throws NonEmptySchemaException If the schema is not empty and cascade is false.
+   * @throws UnsupportedOperationException If the catalog does not support dropping a schema with a
+   *     cascade option.
```

Comment:
What's the purpose of explicit defining of `UnsupportedOperationException`?

Reply:
I think we should provide detailed error messages if some operations are not supported by certain catalogs.